### DISCUSSION TITLE
Bring back ARM_CHANNEL (sort of...)

### DISCUSSION THREAD

**Discussion by padcom** [ExpressLRS/ExpressLRS#1916] [Ideas]
I've been looking for this for ages and found it in obsolete/removed features: changing which channel is used to determine if the aircraft is armed or not. Strictly speaking, after going through the code, I can see the following features being dependent on AUX1 (or rather the `static bool IsArmed() function)` :
- Blocking certain LUA script options when armed (totally unnecessary but doesn't change a thing)
- DynamicPower:
  - when `IsArmed()` == 0 and dynamic power changes are enabled then set it to `MinPower`, otherwise set it `config.GetPower()`
  - when `IsArmed()` == 1 and dynamic changes are enabled, then set it to  `config.GetPower()` function
  - for cases when the connection is lost, the current TX power is automatically set to `config.GetPower()` if `IsArmed() == 1`
  - if AUX channel _NOT_ is set, dynamic power adjustment will use `IsArmed()` to determine if max power should be applied when connection is lost
  - if AUX channel _IS_ set and the AUX value is `off` then the power will be dynamically increased
  - if AUX channel _IS NOT_ set, then the power is managed by the `IsArmed()` (acts exactly as setting the AUX channel for dynamic power) 
- disable configuration updates until the controller is disarmed
- disable LCD configuration options until disarmed (neat, completely unnecessary)
- `NO_ASYNC_ON_ARM` - _performance optimization_

With that in mind, if we just allowed to select which channel to use to read the `IsArmed()` information, we could allow the user to select
a) how often do they need to update the `ARMED` state, 
b) what's more important to be updated every packet? ARM or RTH? It depends on te use-case (MR or FW+LR)

It'd be great to resurrect the ARM_CHANNEL configuration option, but as a really configuration option that _just tells ELRS to enable/disable reading that virtual switch. That is more than enough for all usecases.

**Comment by padcom** [comment]
And just so that we're clear: I wouldn't want to change all the other logic. That can be left as-is. Just the AUX1 channel could be something different.

**Comment by pkendall64** [comment]
Some of the things you list as "unnecessary" are actually very important. The blocked LUA options are done because if they are changed when in flight, armed, they can have catastrophic consequences e.g. changing the switch mode has caused crashes that have destroyed the model. The disabling of the OLED/TFT button when armed was also done because some users fiddle with the button under their fingers while flying and would make inadvertent changes that also caused crashes or failsafes. Also, the SPI/I2C commands to the OLED/TFT are slow and cause problems with the business of doing the radio link. The very first release of the OLED code which BETAFPV did had this problem and it caused a lot of problems for their initial users.

You have also missed the devVTX code which does not send VTX MSP commands to the FC when armed. This is also very important and changing your video in flight could be a serious problem.

And also the bump sensing code and inactivity low-power mode for modules that have an accelerometer like the Axis module.

Whilst the devs are keenly aware that not all users need an Arm channel, we do need a way to control these features on TX modules and tying it to the Arm state (for quads) is just a sensible thing to do. We have already removed so "armed" code from the RX side of things and that was one of the things holding us back from allowing the Arm channel to be configured. There have been discussions (and they continue) as to how/if we can remove the armed state knowledge completely from the system.

**Comment by padcom** [comment]
@pkendall64 I am sorry if I didn't make myself clear enough. I am *NOT* stating those features are unnecessary. Far from it! If you'd be kind enough to read the title again you will notice that I am merely proposing the AUX1 be configurable. All those things are cool and useful

It is just a difference between being able to *always* and with *high reseliecy* switch on or off either arming (mostly racers) or engage RTH (everybody else). Believe me, when you're 20 clicks out, dropping supplies, the last thing you want is for the aircraft to disengage. RTH *is* to LR what arming is for racers. You wanna drop it and get outta there ASAP. 

With just a tiny effort that great feature would be within our reach.

**Reply by pkendall64** [reply]
I do realise that you are saying it should be configurable, but it would actually be helpful to know why ARM on channel 5 is such a problem, rather than just saying it should be configurable.

**Reply by scooper1** [reply]
channel 5 on most helicopters is the gyro mode select switch
on planes its the gear channel

**Comment by padcom** [comment]
Oh, and before I forget, check out the Local tab in the configurator - you'll find it there

**Reply by pkendall64** [reply]
Yeah, I know it's there. It's left there so you can build old versions that had that feature.

**Reply by padcom** [reply]
And for all intents and purposes, all it takes is to do it is to make https://github.com/ExpressLRS/ExpressLRS/blob/master/src/lib/CRSF/CRSF.h#L85 read if the value is, let's say, 25% or more (or even make it configurable, why not?!) and to know which channel to read and to resurrect the `ARM_CHANNEL` config option in the configurator.

**Reply by pkendall64** [reply]
The arm channel configuration has not actually been around since version 0.1.3.
Now you say make the channel and the level at which it switches configurable, why not above or below as well? You see how this explodes and becomes a configuration nightmare for users. When we already run into problems with users not realising that it must be channel 5 high already!

**Reply by padcom** [reply]
OK, maybe it's not that super easy, there are tests that would need to be updated too (`test_switches.cpp`) but that should not be a big deal. A small refactoring to extract the writing of `ChannelData[4]` into a meaningful function (like `IsArmed()` does).

**Reply by padcom** [reply]
A way more sophisticated option would be to make it runtime-configurable (via the LUA script), or even model-matched :D Man, that'd be a bomb (in a manner of speaking)!

**Reply by padcom** [reply]
I just read your replay about the min/max... well... IMHO that is way overboard. Keep it simple, event just compile-time configurable like it was before, keep it 50%+ as on (just the way it is right now) and that'll solve the most use cases. Progress - not perfection :)

**Reply by padcom** [reply]
> You have also missed the devVTX code which does not send VTX MSP commands to the FC when armed. This is also very important and changing your video in flight could be a serious problem.

Isn't that read using the `IsArm()`? If not, I bet it not only could but should be refactored to use it. But you're right, I wasn't even looking for it.

Well, for what it is worth, I've been doing it over and over again before, changing channels and power levels mid-flight to understand better which channel is clear enough. Of course, I keep my ARM on AUX2 (3-pos switch) and RTH on AUX1 so unless I am coming back home I can do anything, and it works like a charm. Doing that mid-race would cause we-all-know-what, but seriously, is ELRS just for racing? Should we start another firmware that's just for LR and make it so that it's specifically dangerous to use for racing?

I think there's room for every scenario here. I'd understand if it'll be decided that ELRS will forever only support ARM on AUX1. That's cool too. A fork is always an option.
